sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -591,6 +591,12 @@ def some_pipeline():
       if container:
         self.assertEqual(template['retryStrategy']['limit'], 5)
 
+  def test_withitem_basic(self):
```

Comment:
Is it possible to test the actual behavior of the feature instead of just comparing the YAML text?

It would be great if the tests are not starting to fail when some unrelated part (e.g. pipeline name or metadata) changes.

Reply:
i agree, it'd be nice to have an e2e test as well, but either way, we'll want to include some unit tests too and this is the pattern the repo uses.

Reply:
Unit tests are important and we should have them. However we can usually do better than just comparing YAML output of the whole pipeline (although, checking for the loop compilation behavior might be non-trivial).
As and example of unit tests that test the feature behavior, see 
[test_init_container](https://github.com/kubeflow/pipelines/blob/cd4490554dc8fc67491b60ad0e4ffabe1e84b872/sdk/python/tests/compiler/compiler_tests.py#L614)
[test_op_transformers](https://github.com/kubeflow/pipelines/blob/cd4490554dc8fc67491b60ad0e4ffabe1e84b872/sdk/python/tests/compiler/compiler_tests.py#L589)
[test_set_display_name](https://github.com/kubeflow/pipelines/blob/cd4490554dc8fc67491b60ad0e4ffabe1e84b872/sdk/python/tests/compiler/compiler_tests.py#L551)

Reply:
I agree, less brittle tests would be nice, though the full YAML comparison is more thorough and that is most of how we currently test the compiler.